### DISCUSSION TITLE
Why ImageMagick is taking forever for large ai/pdf files?

### DISCUSSION THREAD

**Discussion by anirbanmalick** [ImageMagick/ImageMagick#3696] [Development]
I am using _**convert -density 600 -resize 640X640 input_image.ai[0] output_image.png**_

While ai file size is up to 20MB, convert works perfectly. It produces result in 1-2 seconds. However, when I am using files >25MB convert command takes forever. Even after waiting for 30 mins I got no output. I tried to reduce density value up to 50. Even after that convert is not working.

Currently I am using ImageMagick7 on Ubuntu 18.04 LTS.

![IM](https://user-images.githubusercontent.com/13747902/118688478-b847a000-b823-11eb-9fa8-42d6fc41c6c2.PNG)

**Comment by fmw42** [comment]
You are likely running out of RAM. Your density is too big for your AI dimensions. You might be able to increase the resources allocated to ImageMagick by editing your policy.xml file. See https://imagemagick.org/script/architecture.php#tera-pixel and https://imagemagick.org/script/resources.php

**Reply by anirbanmalick** [reply]
This is helpful indeed! Thanks.

**Comment by snibgo** [comment]
As fmw42 says.

The size of the .ai file in megabytes has no relevance. The critical size is the number of pixels. This depends on the density (eg pixels per inch) and the images dimensions (in inches).

**Comment by fmw42** [comment]
P.S. Your syntax is not proper, though IM 6 is forgiving. But IM 7 would not be. Proper syntax would read the AI file before the resize. So

`convert -density 600 input_image.ai[0] -resize 640X640 output_image.png`
